app/packs/src/apps/mydb/elements/details/reactions/ReactionDetails.js

```diff
@@ -302,75 +300,92 @@ export default class ReactionDetails extends Component {
-          <span><i className="icon-reaction" />&nbsp;{reaction.title()}</span>
+          <span><i className="icon-reaction me-1" />{reaction.title()}</span>
         </OverlayTrigger>
-        <ConfirmClose el={reaction} />
-        <OverlayTrigger placement="bottom"
-          overlay={<Tooltip id="saveReaction">Save and Close Reaction</Tooltip>}>
-          <Button
-            variant="warning"
-            size="sm"
-            onClick={() => this.handleSubmit(true)}
-            disabled={!permitOn(reaction) || !this.reactionIsValid() || reaction.isNew}
-            style={{ display: hasChanged }}
-          >
-            <i className="fa fa-floppy-o" />
-            <i className="fa fa-times" />
-          </Button>
-        </OverlayTrigger>
-        <OverlayTrigger placement="bottom"
-          overlay={<Tooltip id="saveReaction">Save Reaction</Tooltip>}>
-          <Button variant="warning" size="sm"
-            onClick={() => this.handleSubmit()}
-            disabled={!permitOn(reaction) || !this.reactionIsValid()}
-            style={{ display: hasChanged }} >
-            <i className="fa fa-floppy-o "></i>
-          </Button>
-        </OverlayTrigger>
-        {copyBtn}
-        <OverlayTrigger
-          placement="bottom"
-          overlay={<Tooltip id="fullSample">FullScreen</Tooltip>}
-        >
-          <Button
-            variant="info"
-            size="sm"
-            onClick={() => this.props.toggleFullScreen()}
-          >
-            <i className="fa fa-expand" />
-          </Button>
-        </OverlayTrigger>
-        <OverlayTrigger
-          placement="bottom"
-          overlay={<Tooltip id="generateReport">Generate Report</Tooltip>}
-        >
-          <Button
-            variant="success"
-            size="sm"
-            disabled={reaction.changed || reaction.isNew}
-            title={(reaction.changed || reaction.isNew) ?
-              "Report can be generated after reaction is saved."
-              : "Generate report for this reaction"}
-            onClick={() => Utils.downloadFile({
-              contents: "/api/v1/reports/docx?id=" + reaction.id,
-              name: reaction.name
-            })}
-          >
-            <i className="fa fa-cogs" />
-          </Button>
-        </OverlayTrigger>
-        <div style={{ display: "inline-block", marginLeft: "10px" }}>
+
+        <div className='ms-1'>
           {colLabel}
           {rsPlanLabel}
           <ElementAnalysesLabels element={reaction} key={reaction.id + "_analyses"} />
           <HeaderCommentSection element={reaction} />
         </div>
-        {reaction.isNew
-          ? null
-          : <OpenCalendarButton isPanelHeader eventableId={reaction.id} eventableType="Reaction" />}
-        <PrintCodeButton element={reaction} />
+        <div className='d-flex align-items-center gap-2'>
+          <div className='d-flex flex-row-reverse'>
+            <ConfirmClose el={reaction} />
+            {reaction.changed ?
```

Comment:
With `{reaction.changed && (<>…</>)}` we could get rid of the `null` branch.

There is another instance of this pattern further down as well.